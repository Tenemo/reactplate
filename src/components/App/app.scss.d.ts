// Automatically generated by @teamsupercell/typings-for-css-modules-loader.
// Please do not edit this file manually.
declare namespace AppScssNamespace {
  export interface IAppScss {
    app: string;
    error: string;
    errorInfo: string;
  }
}

declare const AppScssModule: AppScssNamespace.IAppScss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: AppScssNamespace.IAppScss;
};

export = AppScssModule;
