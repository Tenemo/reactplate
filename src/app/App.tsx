import 'normalize.css';

import 'fonts/RobotoMono-Regular.woff2';
import 'fonts/RobotoMono-Regular.woff';

import { Helmet } from '@dr.pogodin/react-helmet';
import React, { useEffect } from 'react';
import { ErrorBoundary } from 'react-error-boundary';
import { Route, Routes } from 'react-router-dom';

import styles from './app.module.scss';

import { useAppSelector } from 'app/hooks';
import Header from 'components/Header/Header';
import NotFound from 'components/NotFound/NotFound';
import { selectAppTheme } from 'features/AppTheme/appThemeSlice';
import HomePage from 'features/HomePage/HomePage';

const App = (): React.JSX.Element => {
    const appTheme = useAppSelector(selectAppTheme);
    const classNames = `${styles.app} ${
        appTheme === 'dark' ? 'theme-dark' : 'theme-light'
    }`;

    useEffect(() => {
        document.documentElement.classList.remove('theme-dark', 'theme-light');
        document.documentElement.classList.add(
            appTheme === 'dark' ? 'theme-dark' : 'theme-light',
        );
    }, [appTheme]);

    return (
        <div className={classNames}>
            <Helmet>
                <title>reactplate</title>
            </Helmet>
            <ErrorBoundary
                fallback={
                    <div>
                        The application has crashed due to a rendering error.
                    </div>
                }
                onError={(error) => {
                    console.error(error);
                }}
            >
                <Header />
                <Routes>
                    <Route element={<HomePage />} path="/" />
                    <Route element={<NotFound />} path="*" />
                </Routes>
            </ErrorBoundary>
        </div>
    );
};

export default App;
