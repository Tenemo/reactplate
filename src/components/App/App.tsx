import React, { Component, ReactElement } from 'react';
import { Route, Switch } from 'react-router-dom';
import 'normalize.css';
import { Helmet } from 'react-helmet-async';

import 'fonts/RobotoMono-Regular.woff2';
import 'fonts/RobotoMono-Regular.woff';

import NotFound from 'components/NotFound';
import Header from 'components/Header';
import HomePage from 'components/HomePage';

import styles from './app.scss';

export class App extends Component {
    render(): ReactElement {
        return (
            <div className={styles.app}>
                <Helmet>
                    <title>Reactplate</title>
                </Helmet>
                <Header />
                <Switch>
                    <Route component={HomePage} exact path="/" />
                    <Route component={NotFound} />
                </Switch>
            </div>
        );
    }
}

export default App;
