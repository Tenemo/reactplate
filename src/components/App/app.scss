.app {
    @font-face {
        font-family: "Roboto Mono";
        src: url("../../fonts/RobotoMono-Regular.woff2") format("woff2"),
            url("../../fonts/RobotoMono-Regular.woff") format("woff");
    }
    font-family: "Roboto Mono", "Courier New", "Courier", monospace;
    padding: 10px 15px;
    height: 100%;
    display: flex;
    flex-direction: column;
}
:global .theme-dark {
    color: #ddd;
    background-color: #222;
}
:global .theme-light {
    color: #222;
    background-color: #eee;
}
.error {
    color: darkred;
    margin: 10px 20px;
}
.errorInfo {
    color: #aaa;
    margin-top: 20px;
}
