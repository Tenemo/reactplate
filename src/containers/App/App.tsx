import React, { Component } from 'react';
// import { Route, Switch, Redirect } from 'react-router-dom';
import 'normalize.css';
import 'antd/dist/antd.css';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { ThemeProvider } from 'styled-components';
// import NotFound from 'components/NotFound'

const theme = {
    main: 'black',
};

export class App extends Component {
    render() {
        return (
            <ThemeProvider theme={theme}>
                <React.Fragment>
                    <div>Reactplate</div>
                    <p>Reactplate</p>
                </React.Fragment>
            </ThemeProvider>
        );
    }
}

export default App;
