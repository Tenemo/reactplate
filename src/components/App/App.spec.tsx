import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe(`App`, (): void => {
    it(`should match the snapshot`, (): void => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    });
});
