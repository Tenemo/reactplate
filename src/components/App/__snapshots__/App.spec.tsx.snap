// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Reactplate
    </title>
  </Helmet>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
